[package] pkgconf/1.7.3: undefined strlcpy/strlcat then building with clang 10

### Package and Environment Details (include every applicable attribute)
  * Package Name/Version: **pkgconf/1.7.3**
  * Operating System+version: **Fedora 32**
  * Compiler+version: **clang 10**
  * Docker image: **docker.io/fedora:32**
  * Conan version: **1.35.2**


### Conan profile (output of `conan profile show default` or `conan profile show <profile>` if custom profile is in use)
```
+---------------------+
| Profile             |
|---------------------|
| include(clang)      |
|                     |
| [settings]          |
| arch=x86_64         |
| build_type=Release  |
| compiler=clang      |
| compiler.version=10 |
| [options]           |
| pkgconf:shared=True |
| [env]               |
|                     |
| [build_requires]    |
+---------------------+
```


### Steps to reproduce (Include if Applicable)



### Logs (Include/Attach if Applicable)
<details><summary>Click to expand log</summary>

```
pkgconf/1.7.3: Calling build()
The Meson build system
Version: 0.57.2
Source dir: /root/.conan/data/pkgconf/1.7.3/_/_/build/da82c45faf95f16de2c6a6f3b0d205907ceadfee/source_subfolder
Build dir: /root/.conan/data/pkgconf/1.7.3/_/_/build/da82c45faf95f16de2c6a6f3b0d205907ceadfee/build_subfolder
Build type: native build
Project name: pkgconf
Project version: 1.7.3
C compiler for the host machine: clang (clang 10.0.1 "clang version 10.0.1 (Fedora 10.0.1-3.fc32)")
C linker for the host machine: clang ld.bfd 2.34-6
Host machine cpu family: x86_64
Host machine cpu: x86_64
Has header "dlfcn.h" : YES 
Has header "inttypes.h" : YES 
Has header "memory.h" : YES 
Has header "stdint.h" : YES 
Has header "stdlib.h" : YES 
Has header "strings.h" : YES 
Has header "string.h" : YES 
Has header "sys/stat.h" : YES 
Has header "sys/types.h" : YES 
Has header "unistd.h" : YES 
Checking for function "cygwin_conv_path" : NO 
Header <sys/cygwin.h> has symbol "cygwin_conv_path" : NO 
Checking for function "strlcat" : NO 
Header <string.h> has symbol "strlcat" : YES 
Checking for function "strlcpy" : NO 
Header <string.h> has symbol "strlcpy" : YES 
Checking for function "strndup" : YES 
Configuring config.h using configuration
Build targets in project: 2
Found ninja-1.10.2 at /root/.conan/data/ninja/1.10.2/_/_/package/24647d9fe8ec489125dfbae4b3ebefaf7581674c/bin/ninja
ninja: Entering directory `/root/.conan/data/pkgconf/1.7.3/_/_/build/da82c45faf95f16de2c6a6f3b0d205907ceadfee/build_subfolder'
[1/20] Compiling C object libpkgconf.so.3.0.0.p/libpkgconf_audit.c.o
[2/20] Compiling C object libpkgconf.so.3.0.0.p/libpkgconf_bsdstubs.c.o
../source_subfolder/libpkgconf/bsdstubs.c:126:9: warning: implicitly declaring library function 'strlcpy' with type 'unsigned long (char *, const char *, unsigned long)' [-Wimplicit-function-declaration]
        return strlcpy(dst, src, siz);
               ^
../source_subfolder/libpkgconf/bsdstubs.c:126:9: note: include the header <string.h> or explicitly provide a declaration for 'strlcpy'
../source_subfolder/libpkgconf/bsdstubs.c:132:9: warning: implicitly declaring library function 'strlcat' with type 'unsigned long (char *, const char *, unsigned long)' [-Wimplicit-function-declaration]
        return strlcat(dst, src, siz);
               ^
../source_subfolder/libpkgconf/bsdstubs.c:132:9: note: include the header <string.h> or explicitly provide a declaration for 'strlcat'
2 warnings generated.
[3/20] Compiling C object libpkgconf.so.3.0.0.p/libpkgconf_fileio.c.o
[4/20] Compiling C object libpkgconf.so.3.0.0.p/libpkgconf_cache.c.o
[5/20] Compiling C object libpkgconf.so.3.0.0.p/libpkgconf_argvsplit.c.o
[6/20] Compiling C object libpkgconf.so.3.0.0.p/libpkgconf_parser.c.o
[7/20] Compiling C object pkgconf.p/cli_renderer-msvc.c.o
[8/20] Compiling C object libpkgconf.so.3.0.0.p/libpkgconf_path.c.o
[9/20] Compiling C object libpkgconf.so.3.0.0.p/libpkgconf_queue.c.o
[10/20] Compiling C object libpkgconf.so.3.0.0.p/libpkgconf_dependency.c.o
[11/20] Compiling C object libpkgconf.so.3.0.0.p/libpkgconf_personality.c.o
[12/20] Compiling C object libpkgconf.so.3.0.0.p/libpkgconf_tuple.c.o
[13/20] Compiling C object libpkgconf.so.3.0.0.p/libpkgconf_client.c.o
[14/20] Compiling C object pkgconf.p/cli_getopt_long.c.o
[15/20] Compiling C object libpkgconf.so.3.0.0.p/libpkgconf_fragment.c.o
[16/20] Compiling C object pkgconf.p/cli_main.c.o
[17/20] Compiling C object libpkgconf.so.3.0.0.p/libpkgconf_pkg.c.o
[18/20] Linking target libpkgconf.so.3.0.0
FAILED: libpkgconf.so.3.0.0 
clang  -o libpkgconf.so.3.0.0 libpkgconf.so.3.0.0.p/libpkgconf_argvsplit.c.o libpkgconf.so.3.0.0.p/libpkgconf_audit.c.o libpkgconf.so.3.0.0.p/libpkgconf_bsdstubs.c.o libpkgconf.so.3.0.0.p/libpkgconf_cache.c.o libpkgconf.so.3.0.0.p/libpkgconf_client.c.o libpkgconf.so.3.0.0.p/libpkgconf_dependency.c.o libpkgconf.so.3.0.0.p/libpkgconf_fileio.c.o libpkgconf.so.3.0.0.p/libpkgconf_fragment.c.o libpkgconf.so.3.0.0.p/libpkgconf_parser.c.o libpkgconf.so.3.0.0.p/libpkgconf_path.c.o libpkgconf.so.3.0.0.p/libpkgconf_personality.c.o libpkgconf.so.3.0.0.p/libpkgconf_pkg.c.o libpkgconf.so.3.0.0.p/libpkgconf_queue.c.o libpkgconf.so.3.0.0.p/libpkgconf_tuple.c.o -Wl,--as-needed -Wl,--no-undefined -Wl,-O1 -shared -fPIC -Wl,--start-group -Wl,-soname,libpkgconf.so.3 -Wl,--end-group -m64 -m64 -O3 -DNDEBUG
/usr/bin/ld: libpkgconf.so.3.0.0.p/libpkgconf_bsdstubs.c.o: in function `pkgconf_strlcpy':
bsdstubs.c:(.text+0x1): undefined reference to `strlcpy'
/usr/bin/ld: libpkgconf.so.3.0.0.p/libpkgconf_bsdstubs.c.o: in function `pkgconf_strlcat':
bsdstubs.c:(.text+0x11): undefined reference to `strlcat'
clang-10: error: linker command failed with exit code 1 (use -v to see invocation)
ninja: build stopped: subcommand failed.
pkgconf/1.7.3: 
pkgconf/1.7.3: ERROR: Package 'da82c45faf95f16de2c6a6f3b0d205907ceadfee' build failed
pkgconf/1.7.3: WARN: Build folder /root/.conan/data/pkgconf/1.7.3/_/_/build/da82c45faf95f16de2c6a6f3b0d205907ceadfee
```

</details>

The problem is the patch for clang 12 added in  #5465, which results in a false positive detection of `strlcpy`/`strlcat`.
If I simply remove the `clang-12-strndup-not-in-string-h` patch, building works fine.

Currently unsure how this can be fixed to work with both clang 10 and 12.

/cc @madebr

--- a/recipes/pkgconf/all/patches/1.7.3-0002-clang-12-strndup-not-in-string-h.patch
+++ b/recipes/pkgconf/all/patches/1.7.3-0002-clang-12-strndup-not-in-string-h.patch
@@ -5,7 +5,7 @@
  
  foreach f : check_functions
 -  if cc.has_function(f.get(1), prefix : '#include <' + f.get(2) + '>') and cc.has_header_symbol(f.get(2), f.get(1))
-+  if cc.has_function(f.get(1), prefix : '#include <' + f.get(2) + '>') or cc.has_header_symbol(f.get(2), f.get(1))
++  if cc.has_function(f.get(1)) or (cc.has_function(f.get(1), prefix : '#include <' + f.get(2) + '>') and cc.has_header_symbol(f.get(2), f.get(1)))
      cdata.set(f.get(0), 1)
    endif
  endforeach

--- a/recipes/pkgconf/all/patches/1.7.4-0001-clang-12-strndup-not-in-string-h.patch
+++ b/recipes/pkgconf/all/patches/1.7.4-0001-clang-12-strndup-not-in-string-h.patch
@@ -5,7 +5,7 @@
  
  foreach f : check_functions
 -  if cc.has_function(f.get(1), prefix : '#include <' + f.get(2) + '>') and cc.has_header_symbol(f.get(2), f.get(1))
-+  if cc.has_function(f.get(1), prefix : '#include <' + f.get(2) + '>') or cc.has_header_symbol(f.get(2), f.get(1))
++  if cc.has_function(f.get(1)) or (cc.has_function(f.get(1), prefix : '#include <' + f.get(2) + '>') and cc.has_header_symbol(f.get(2), f.get(1)))
      cdata.set(f.get(0), 1)
    endif
  endforeach

